internal/clients/api_client.go

```diff
@@ -214,82 +223,112 @@ func newEsApiClient(d *schema.ResourceData, key string, version string, useEnvAs
 					for _, e := range strings.Split(endpoints, ",") {
 						addrs = append(addrs, strings.TrimSpace(e))
 					}
-					config.Addresses = addrs
+					baseConfig.Addresses = addrs
 				}
 			}
 
-			if endpoints, ok := esConfig["endpoints"]; ok && len(endpoints.([]interface{})) > 0 {
+			if endpoints, ok := config["endpoints"]; ok && len(endpoints.([]interface{})) > 0 {
```

Comment:
As before, the endpoints aren't shared. We should probably move this logic into the newEsClient func instead and just have the credentials in the shared config.